### ISSUE TITLE
Dynamic Forward Proxy: No endpoint added to cluster with same IP but different port 

### ISSUE THREAD

**Issue by Coconut105** [envoyproxy/envoy#15441]
Hi,

I have configured a cluster that uses dynamic forward proxy and also EDS. If I send requests to envoy with the same FQDN but different ports, only the first resolved IP and port are added in the endpoints of the cluster. Eg:

1) Request: google.com:80  -> Endpoint: 10.0.0.1:80 added to cluster
2) Request: google.com:8081 -> 503 Service unavailable

In the envoy logs I can see a cache hit for the second request and then request is sent to endpoint with port 80 instead of 8081. The second endpoint is not added to the cluster. I believe this is a bug.

**Comment by ntgsx92** [comment]
I encountered similar situation while trying out DynamicForwardProxy filter. The underlying DNS cache used by the filter will only use either [port 80 or 443](https://github.com/envoyproxy/envoy/blob/80c0abdc9b4491fa1c35bc20577bb638147cff09/source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc#L82-L86) depends on whether TLS connection is needed.

I think the reason why this was implemented this way is that Envoy currently doesn't support [SRV record lookup](https://github.com/envoyproxy/envoy/issues/125) from the DNS server which means no port number will be returned. Once this issue is resolved, we should be able to use IP address plus the port returned from the DNS server and establish connection with the clusters.
